### DISCUSSION TITLE
Questions - Community and Contributors

### DISCUSSION THREAD

**Discussion by ehsan6sha** [functionland/fx-fotos#58] [Q&A]
Users, developers and contributors, feel free to ask any question you might have about Photos project.

**Comment by emadbaqeri** [comment]
Hey, how are we supposed to name branched when we are working on an issue.
For issue Module Renaming #12 I'm going to use Module_Ranaming#12. 
Am I right? If not please correct me.

**Reply by ehsan6sha** [reply]
I think That looks perfect

**Reply by emadbaqeri** [reply]
I'm glad hearing that. Thank you :)

**Comment by emadbaqeri** [comment]
I start to setup the project with README file but there is an error below that belames about BigInt. Do you know what is the problem with this? I was not able to track the error.

![Screenshot_1626206452](https://user-images.githubusercontent.com/50639499/125517289-890afc59-0f43-4aea-b88a-d2251da5d150.png)

**Reply by ehsan6sha** [reply]
> I start to setup the project with README file but there is an error below that belames about BigInt. Do you know what is the problem with this? I was not able to track the error.
> 
> ![Screenshot_1626206452](https://user-images.githubusercontent.com/50639499/125517289-890afc59-0f43-4aea-b88a-d2251da5d150.png)

Ah yes, I forgot to mention. We are adding Dfinity support and right now their package does not support react native. So we did custom polyfill to their libraries. To make it work, you need to replace the following files after you ran yarn (and you need to do this if you run yarn again, not optimized but working to bake it into code):
(temp/ files mentioned are here: https://github.com/functionland/photos/tree/main/temp)

1- copy and replace "temp/transforms.js" to "...\Photos\node_modules\@dfinity\agent\lib\cjs\agent\http\transforms.js"
2- copy and replace "temp/index.js" to "...\Photos\node_modules\@dfinity\authentication\lib\cjs\index.js"
3- copy and replace "temp/index.d.ts" to "...\Photos\node_modules\@dfinity\auth-client\lib\cjs\index.d.ts"
4- copy and replace "temp/index-2/index.js" to "...\Photos\node_modules\@dfinity\auth-client\lib\cjs\index.js"
5- copy and replace "temp/index-3/index.js" to "...\Photos\node_modules\@dfinity\principal\lib\cjs\index.js"

Let me know if any issues. I will bake these into the code somehow

**Reply by emadbaqeri** [reply]
Thanks for your help I will do it and inform you as soon as I can

**Comment by emadbaqeri** [comment]
Another question is about static testing using eslint and code formatting configuration. Is there any shared configuration for the developers working on this repository. I'm using emacs for development and code formatting rules are different than the thing used in the this repository. I think it may cuase some issue when merging the changes from my branch to the main.
Sorry for asking really lot's of questions but I really need to opensource I love this community.

**Reply by ehsan6sha** [reply]
> Another question is about static testing using eslint and code formatting configuration. Is there any shared configuration for the developers working on this repository. I'm using emacs for development and code formatting rules are different than the thing used in the this repository. I think it may cuase some issue when merging the changes from my branch to the main.
> Sorry for asking really lot's of questions but I really need to opensource I love this community.

Very good point actually and thanks for mentioning it. I created a .editorconfig and committed it under main branch. Although right now not all files might be consistent but aim is to make them consistent.

**Reply by emadbaqeri** [reply]
how can I use .editorconfig file in vscode. I'm really new to it but it looks amazing. it's been few days I'm using it, there are lots of good looking and fancy themes but it's not as customizable as other oldschool IDEs.

**Comment by emadbaqeri** [comment]
A question come to my mind. while I'm running the project there are lots of logs in the console. Are they really neccessary for the development or we should just clean them up.

**Reply by ehsan6sha** [reply]
I know :D Some of them are warnings due to some of 3rd parties. TBH I kept them to have them in my head to check the code and correct the cause of those

**Reply by emadbaqeri** [reply]
I know that but there are lot's of the on the console.log('example text') in the code, and logging from the different useEffects and event handler function

**Comment by ehsan6sha** [comment]
@emadbaqeri can you please help @gitaaron to set this up? It would be a good practice and we can then document it.
Aaron said: "I was able to get the react-native playground working with iOS simulator, however, it seemed to crash when I tried the Photos app"

**Reply by emadbaqeri** [reply]
Hey @ehsan6sha, surely.
Aaron, could you please tell me how can I reproduce the issue on my machine?

**Reply by gitaaron** [reply]
@ehsan6sha -  it seems that Photos is not currently working with the iOS simulator.  Would you like me to try and get it working with an android emulator?  Last time we tried we ran issues because I am running with apple m1 but @emadbaqeri says he has a fix for that.

**Comment by DevilsDesigns** [comment]
is there a feature for videos that will be a feature in the near future. I love your app so far. But I take a lot of videos and this would work great if it had that option.

**Reply by ehsan6sha** [reply]
Thank you @DevilsDesigns for the encouraging comment! Definitely! Even now, it is not just a Photos app, it is a gallery for both Photos and Videos with the interface of big gallery apps from cloud providers. 

Also since you mentioned taking video a lot, It is backend agnostic but we are currently developing a novel backend for it as well (under Fula repo) that is a new category of product called Blockchain-Attached Storage. You can find more information on our Indiegogo presale page if you want: https://igg.me/at/functionland (You can even build it yourself on your computer or with a raspberry pi)

**Reply by DevilsDesigns** [reply]
That's would be awesome I have 2 servers with both 16+TB of NAS drives. I would love to locally run this on my windows servers is that possible using a windows 10 install

**Reply by ehsan6sha** [reply]
> That's would be awesome I have 2 servers with both 16+TB of NAS drives. I would love to locally run this on my windows servers is that possible using a windows 10 install

yes!! https://docs.fx.land shows the process. However, the encryption layer is still in development. So use for testing purposes

**Reply by DevilsDesigns** [reply]
is there a way to run locally without docker?

**Reply by gitaaron** [reply]
> is there a way to run locally without Docker?

HI @DevilsDesigns you only need to use Docker if you want to play around with the Box app - the back-end platform that Fotos will integrate with.  Are you interested in playing around with the Fotos app?  If so you should be able to get it running without having to work about getting it working with the back-end.

**Reply by gitaaron** [reply]
If you are interested in getting the Box app there is a [way](https://github.com/functionland/fula#local-development-on-host-os) to get it running without using Docker on a host OX using [Rush](https://rushstack.io/)

**Reply by DevilsDesigns** [reply]
I am going to attempt to get it working on windows 10. Would you like me to post a tutorial on how to get it working on your wiki with a git pull?

**Reply by DevilsDesigns** [reply]
if i am succesfull

**Reply by ehsan6sha** [reply]
@DevilsDesigns Video player is added. We are working on having more features for video like:
1- Picture in Picture which allows you to browse while playing a video
2- Video Edit
3- Background detection and removal
4- on-device contacts' faces detection and tagging

**Comment by dakshbhardwaj** [comment]
Hello @ehsan6sha, I am facing some issues while running the project on iOS. I have followed all the steps mentioned in the `readMe` file.  Do I need to do any additional steps ?

<image src="https://user-images.githubusercontent.com/22423684/179986782-ff582a12-fdf3-4c35-93e6-f59d2d723337.png" height="600" width="300"/>

**Reply by ghorbani-m** [reply]
Hi @dakshbhardwaj, Now we are working on it to support react-native-fula on IOS, it is coming soon.
We have a [PR](https://github.com/functionland/react-native-fula/pulls) to fix this issue.
For now, you can get the react-native-fula package from my [repo](https://github.com/ghorbani-m/react-native-fula) to solve this issue.

Also, there is another [PR](https://github.com/functionland/fotos/pull/228) on Fotos repo to prepare the IOS build's configs.

**Reply by ehsan6sha** [reply]
@dakshbhardwaj iOS should now work. we also released it on app store under "Fx Fotos"

**Comment by dipanshunagar** [comment]
Hi @ehsan6sha, just came across the project, loved the aim and concept, yet to try it out!
Is it possible to integrate Fotos with a more traditional self-hosting backend like Nextcloud?

**Reply by ehsan6sha** [reply]
Thanks for the encouraging words. Definitely! We are making it backend agnostic and already created a version that works with Dfinity backend. We will release first version on google play in a month or so and we really appreciate community contribution for other backend.

**Reply by ehsan6sha** [reply]
For this we need to release plugins for other backends and we will start on it as soon as we complete the Fula backend

**Reply by dipanshunagar** [reply]
Would love to see that, thanks! I am mostly a python backend dev, would love to contribute if that skill fits anywhere in the future!

**Reply by mrkrehman** [reply]
Yes, I'm also looking for either my own hosting platform or any other plugin like said above.

**Reply by ehsan6sha** [reply]
@mrkrehman @dipanshunagar 
We are working on Fula backend integration right now. This should be released by end of the year and it should give you the below options. I think you are more interested in option 1:
1- Install Fula backend on any hardware you have (even a raspberry pi 4 works very well). You can even have multiple backends and they all get synced automatically.
2- Connect to Fula network and use storage of and provide storage to others. If you provide storage to others you can earn Fula tokens which you can use toward getting storage from others (and have backups of your data, all encrypted, on other peoples' hardware)

Also the application is now released (without backend)  on Both Google Play and App store under "Fx Fotos" name
